config/azure/build.yaml

```diff
@@ -0,0 +1,71 @@
+# Standard build + test job
+
+parameters:
+  name: ''
+  vmImage: ''
+  container: ''
+  threads: 2
+
+jobs:
+
+ - job: ${{ parameters.name }}
+
+   ${{ if ne(parameters.container, '') }}:
+      container: ${{ parameters.container }}
+
+   strategy:
+     matrix:
+       Release:
+         BUILD_TYPE: 'RELEASE'
+       Debug:
+         BUILD_TYPE: 'DEBUG'
+
+   pool:
+     vmImage: ${{ parameters.vmImage }}
+
+   variables:
+     ARNOLD_ROOT: $(System.DefaultWorkingDirectory)/arnoldRoot
+     DELIGHT: $(System.DefaultWorkingDirectory)/3delight
+
+   steps:
+
+   - script: |
+       brew update &&
+       brew install scons doxygen &&
+       brew cask install xquartz inkscape
+     displayName: 'Install toolchain (Darwin)'
+     condition: eq( variables['Agent.OS'], 'Darwin' )
+
+   - script: |
+       ./config/travis/installDelight.sh &&
+       ./config/azure/installArnold.sh &&
+       ./config/travis/installDependencies.sh
+     displayName: 'Install dependencies'
+     env:
+       ARNOLD_LOGIN: $(arnoldLogin)
+       ARNOLD_PASSWORD: $(arnoldPassword)
+
+   - script: |
+       echo BUILD_TYPE=$BUILD_TYPE
+       g++ --version
+       scons -j ${{ parameters.threads }} install ENV_VARS_TO_IMPORT=PATH BUILD_TYPE=$BUILD_TYPE DELIGHT_ROOT=$DELIGHT ARNOLD_ROOT=$ARNOLD_ROOT BUILD_CACHEDIR=sconsCache
+     displayName: 'Build (Darwin)'
+     condition: eq( variables['Agent.OS'], 'Darwin' )
+
+   - script: |
+       echo BUILD_TYPE=$BUILD_TYPE
+       g++ --version
+       Xvfb :99 -screen 0 1280x1024x24 &
+       metacity&
+       scons -j ${{ parameters.threads }} install ENV_VARS_TO_IMPORT=PATH BUILD_TYPE=$BUILD_TYPE DELIGHT_ROOT=$DELIGHT ARNOLD_ROOT=$ARNOLD_ROOT BUILD_CACHEDIR=sconsCache
```

Comment:
Duplicating the `scons` line between platforms seems undesirable from a DRY perspective. Is this split really because we need `Xvfb` on Linux and don't on Mac? If so, can we just move the `Xvfb` bit into a section of its own? Maybe we should limit ourselves to a single section for mac specifics and a single section for linux specifics?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3217/commits/f72304d3007cf86fdaa67e531d2e9948e9a86b1e